signer/src/stacks/api.rs

```diff
@@ -1476,10 +1327,45 @@ where
                     );
                     break;
                 }
+                // A 404 could also mean there is a bitcoin block with no stacks
+                // blocks anchored.
+                // TODO: can there be other reasons for a 404?
+                if let Error::StacksNodeResponse(ref reqwest_error) = error
+                    && reqwest_error.status() == Some(StatusCode::NOT_FOUND)
```

Comment:
We'll need to update this to deal with the fact that the Fallback client doesn't return the original error.